apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Query/UserRoleQuery.cs

```diff
@@ -36,4 +37,60 @@ public partial class Query
     public async Task<IEnumerable<UserRoleDto>> GetUserRolesAsync(
         [Service] IMarketParticipantClient_V1 client) =>
         await client.UserRolesGetAsync();
+
+    public async Task<IEnumerable<ActorViewDto>> GetUserRoleViewAsync(
+        Guid userId,
+        [Service] IMarketParticipantClient_V1 client)
+        {
+            var usersActors = await client
+            .UserActorsGetAsync(userId)
+            .ConfigureAwait(false);
+
+            var fetchedActors = new List<ActorDto>();
+
+            foreach (var actorId in usersActors.ActorIds)
+            {
+                fetchedActors.Add(await client.ActorGetAsync(actorId).ConfigureAwait(false));
+            }
+
+            var actorViews = new List<ActorViewDto>();
+
+            foreach (var organizationAndActors in fetchedActors.GroupBy(actor => actor.OrganizationId))
+            {
+                var organization = await client
+                    .OrganizationGetAsync(organizationAndActors.Key)
+                    .ConfigureAwait(false);
+
+                foreach (var actor in organizationAndActors)
+                {
+                    var actorUserRoles = new List<UserRoleViewDto>();
+                    var assignedRoles = await client
+                        .ActorsUsersRolesGetAsync(actor.ActorId, userId)
+                        .ConfigureAwait(false);
+
+                    var assignmentLookup = assignedRoles
+                        .Select(ar => ar.Id)
+                        .ToHashSet();
+
+                    foreach (var userRole in await client.ActorsRolesAsync(actor.ActorId).ConfigureAwait(false))
+                    {
+                        actorUserRoles.Add(new UserRoleViewDto(
+                            userRole.Id,
+                            userRole.EicFunction,
+                            userRole.Name,
+                            userRole.Description,
+                            assignmentLookup.Contains(userRole.Id) ? actor.ActorId : null));
+                    }
```

Comment:
And this. Perhaps consider refactoring all these nested `foreach` into `selectMany` calls instead.